app/models/assignment.rb

```diff
@@ -752,11 +752,11 @@ def get_num_valid
 
   def get_num_marked(ta_id = nil)
     if ta_id.nil?
-      results_join = groupings.left_outer_joins(:current_result)
+      results_join = groupings.includes(:current_result).where('groupings.is_collected': true)
```

Comment:
It looks like we're only considering collected groupings now, which is good. This means this part can be simplified quite a bit, just doing a filter for results that have marking state 'complete', rather than the current indirect calculation. You still need to keep the `groupings.joins(:current_result)` part (the `:current_result` limits this to only one result per grouping).

Reply:
(and similarly below)

Reply:
@ANUSHIYAPRIYA You can use a `.joins` to simplify this query.

- Results only exist if submissions are collected, so doing a `joins(:current_result)` will make the `.where('groupings.is_collected': true)` unnecessary.
- Below, it will require that the associated result exists, and so the `where('results.id': nil)` can be removed (and you can remove the `.or` too, and just use a single `.where`)